prow/github/app_auth_roundtripper_test.go

```diff
@@ -372,6 +372,32 @@ func TestAppsAuth(t *testing.T) {
 				return nil
 			},
 		},
+		{
+			name:                "Check app installation for repo uses JWT",
+			cachedAppSlug:       utilpointer.String("ci-app"),
+			cachedTokens:        map[int64]*AppInstallationToken{1: {Token: "the-token", ExpiresAt: time.Now().Add(time.Hour)}},
+			cachedInstallations: map[string]AppInstallation{"kuber": {ID: 1}},
+			doRequest: func(c Client) error {
+				_, err := c.IsAppInstalled("kuber", "k8s")
+				return err
+			},
+			responses: map[string]*http.Response{"/repos/kuber/k8s/installation": {
+				StatusCode: 200,
+				Body:       serializeOrDie(AppInstallation{}),
+			}},
+			verifyRequests: func(r []*http.Request) error {
+				if n := len(r); n != 1 {
+					return fmt.Errorf("expected exactly one request, got %d", n)
+				}
+				if val := r[0].Header.Get("Authorization"); !strings.HasPrefix(val, "Bearer ") || val == "Bearer the-token" {
```

Comment:
This test currently verifies that your change does not work, `the-token` is an access token to the `kuber` installation, not a JWT :) The reason for this is that the regex doesn't match correctly.

Reply:
The condition is a little confusing looking, but it's actually checking that the token is **not** "the-token". I can split it into separate conditions if that makes it clearer. I actually verified the correctness here both in the debugger and by pulling these changes into my downstream tool and testing the functionality.

Reply:
Okay, then I am confused as to how this works as the regexp does not match `/repos/kuber/k8s/installation`: https://regex101.com/r/onRTB2/1 ..?

Reply:
The first `\` in each `\\w*` section is just an escape. Try https://regex101.com/r/dUEeQd/1

Reply:
I just updated it to use back ticks instead to make it more legible.